internal/services/compute/shared_image_resource.go

```diff
@@ -318,16 +326,32 @@ func resourceSharedImageCreateUpdate(d *pluginsdk.ResourceData, meta interface{}
 		image.GalleryImageProperties.OsState = compute.OperatingSystemStateTypesGeneralized
 	}
 
-	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.GalleryName, id.ImageName, image)
-	if err != nil {
-		return fmt.Errorf("creating/updating %s: %+v", id, err)
-	}
+	if d.IsNewResource() {
+		// Due to cache issue, creating image sometimes can not find parent gallery even if the gallery can be retrieved successfully, so adding retry for creation
```

Comment:
instead of fixing this here can we move this logic to where we create the image so it doesn't return until 7 gets or something do not return 404? otherwise this is only fixing this here and no everywhere else we consume the shared image.

you can see this sort of logic in the AD provider for creating groups etc